LittleFs_Encrypted_RemoteDisk/README.md

```diff
@@ -0,0 +1,184 @@
+
+## Authenticated Encryption
+
+Authenticated encryption is a form of encryption that supports the use of additional data to authenticate the
+successful decryption of the ciphertext. The generated ciphertext is combined with additional authentication data
+(AAD) to produce an authentication tag, which is stored alongside the ciphertext. When decrypting, this same AAD
+can be used with the decrypted data and authentication tag to verify the data was correctly decrypted. wolfCrypt
+provides a combined ChaCha20-Poly1305 function to encrypt and decrypt data (using ChaCha20 encryption), and to generate
+or verify authentication data (using the Poly1305 authentication algorithm).
+
+In the application, encryption is performed at the level of littlefs filesystem blocks. The backing store consists of a
+pool of fixed-size blocks, each with an associated metadata block used to hold the authentication tag for the encrypted
+block. Each block is encrypted with the generated key/IV values, with the block index used as additional authentication
+data (AAD) for the Poly1305 authentication. 
+
+For more information on ChaCha20-Poly1305 encryption and authentication, see
+[here](https://en.wikipedia.org/wiki/ChaCha20-Poly1305).
+
+## Other notes
+
+The project is based on the [Littlefs Remote project](https://github.com/Azure/azure-sphere-gallery/tree/main/LittleFs_RemoteDisk)
+
+Here's a link to the [Littlefs project on Github](https://github.com/littlefs-project/littlefs).
+
+## Project Contents
+
+| File/folder | Description |
+|-------------|-------------|
+| `src\HighLevelApp`       | Azure Sphere Sample App source code |
+| `src\PyDiskHost`       | Python app that hosts a 4MB Remote Disk |
+| `README.md` | This README file. |
+| `LICENSE.txt`   | The license for the project. |
+
+## Prerequisites & Setup
+
+- An Azure Sphere-based device with development features (see [Get started with Azure Sphere](https://azure.microsoft.com/services/azure-sphere/get-started/) for more information).
+- Setup a development environment for Azure Sphere (see [Quickstarts to set up your Azure Sphere device](https://learn.microsoft.com/azure-sphere/install/overview) for more information).
+
+This project requires the 23.05 or later SDK, and a device running the 23.05 (or later) Azure Sphere OS.
+
+## How to use
+
+### Python Remote Storage app
+The project contains a Python Flask application (PyDiskHost.py) that supports 4MB storage (matching the defined storage
+layout of the high-level Azure Sphere application). The Python application supports HTTP GET (read), and POST (Write) functions.
+The 4MB of storage is represented by an array of 256byte blocks, each with a 16 byte metadata block used for the authentication
+tag.
+
+The storage size is defined on lines 24-27:
+
+```python
+DISK_SIZE = 4 * 1024 * 1024
+BLOCK_SIZE = 256 
+METADATA_SIZE = 16
+BLOCK_COUNT = DISK_SIZE // BLOCK_SIZE
+```
+
+Start the Python application before running the Azure Sphere application.
```

Comment:
note on opening the firewall on windows here?

Reply:
Done